azure-pipelines.yml

```
@@ -240,14 +243,17 @@ jobs:
       cmakeArgs: '..'
 
   - script: |
+      set -eo pipefail
       make clangformat
       git diff
       if [ "$(git diff $(Build.SourceVersion))" != "" ]; then exit 1; fi
```

You can probably replace both calls to `git diff` with:

```suggestion
      git diff --exit-code $(Build.SourceVersion)
```

Nicer to rely on git's exit code than looking at its output.